docs/testclient.md

```diff
@@ -26,6 +26,35 @@ function calls, not awaitables.
 You can use any of `requests` standard API, such as authentication, session
 cookies handling, or file uploads.
 
+For example, to set headers on the TestClient you can do:
+
+```python
+client = TestClient(app)
+
+# Set headers on the client for future requests
+client.headers = {"Authorization": "..."}
+response = client.get("/")
+
+# Set headers for each request separately
+response = client.get("/", headers={"Authorization": "..."})
+```
+
+And for example to send files with the TestClient:
+
+```python
+client = TestClient(app)
+
+# Send a single file
+files = {"file": open("example.txt", "rb")}
+response = client.post("/form", files=files)
+
+# Send multiple files
+files = {"file1": open("example.txt", "rb"), "file2": open("example.png", "rb")}
+response = client.post("/form", files=files)
```

Comment:
With `open()` like this, pytest will warn about unclosed resources, if we set to raise errors on warnings. 🤔

Also, I wrote this comment years ago, which kind of mismatch the written here: https://github.com/tiangolo/fastapi/issues/1536#issuecomment-640781718

Reply:
Yeah that's a good point, but we also have this case in Starlette `test_staticfiles.py` which pytest warns about this. But I agree it's better to use the contextt manager. Let me update..

Reply:
I think this should be ok now, thank you Marcelo.